docs_user/modules/proc_using-new-subnet-ranges.adoc

```diff
@@ -46,59 +26,53 @@ network_config:
       next_hop: 192.168.1.1
     - ip_netmask: 172.31.0.0/24 <1>
       next_hop: 192.168.1.100 <2>
-```
+----
 <1> The new control plane subnet.
 <2> The control plane IP address of the adopted data plane node.
-
-Do the same for other networks that will need to use different subnets for the
-new and old parts of the deployment.
-
-Once done, run `openstack overcloud node provision` to apply the new configuration.
-
-Note that network configuration changes are not applied by default to avoid
-risk of network disruption. You will have to enforce the changes by setting the
++
+Repeat this step for other networks that need to use different subnets for the new and old parts of the deployment.
+
+. Apply the new configuration to every adopted node:
++
+----
+$ openstack overcloud node provision
+--stack <stack>
+--network_config
+----
++
+[NOTE]
+Network configuration changes are not applied by default to avoid
+the risk of network disruption. You must enforce the changes by setting the
 `StandaloneNetworkConfigUpdate: true` in the {OpenStackPreviousInstaller} configuration files.
```

Comment:
Please carefully review steps 1 and 2  and ensure that the order is logical.  Also, the `openstack overcloud node provision` command in step 2 is not complete. Can you help provide a complete command that the customers would run?

Reply:
I am not sure which steps you refer to here, and which command you think is incomplete. Please clarify.

Reply:
@booxter Can you please review lines 18 - 45 in docs_user/modules/proc_using-new-subnet-ranges.adoc? I hope that helps.

Reply:
Ack. I think I did already.

---

As to the incomplete command,

```
openstack overcloud node provision
--stack <stack>
--network_config
```

Indeed this looks weird. I am not very knowledgeable about TripleO. I wonder if @jistr can dig for a more complete example on how to update nodes' network configuration.

Reply:
@jistr  or @hjensas Can you help provide a more complete example of an `overcloud node provision` command?

Reply:
There is an example in OSP 17 docs: https://docs.redhat.com/en/documentation/red_hat_openstack_platform/17.1/html/installing_and_managing_red_hat_openstack_platform_with_director/assembly_provisioning-and-deploying-your-overcloud#proc_manually-configuring-bare-metal-node-hardware-information_ironic_provisioning

Step 10:
```
(undercloud)$ openstack overcloud node provision \
 [--templates <templates_directory> \]
 --stack <stack> \
 --network-config \
 --output <deployment_file> \
 /home/stack/templates/<node_definition_file>
```

The example here is missing the `--output <deployment_file>` argument and the path to input `<node_definition_file>`.

Reply:
It looks like several steps involved to pre-populates routes on `{OpenStackPreviousInstaller}` is missing from this doc? It depends on the NIC config templates used in the OSP17 deployment, but typically the routes would have to be added to to subnets in `network_data.yaml` -> then the `openstack overcloud network provision` command must be used to update the networks and then `openstack overcloud node provision` / `openstack overcloud deploy` commands must re-run.

Reply:
CC: @booxter

Reply:
I'm looking at the latest version of the doc, and I think it was messed by some changes. E.g. this diff is incorrect:

```diff
----
-
-Before proceeding, configure host routes on the adopted nodes for the
-control plane subnets.
-
-To achieve this, you will need to re-run `openstack overcloud node provision` with additional `routes` entries added to `network_config`. (This change should be applied for every adopted node configuration.) For example, you may add the following to `net_config.yaml`:
-
-```yaml
+. In the `NetworkAttachmentDefinition` CR, configure host routes on the adopted nodes for the control plane subnets. For example:
++
+----
```

which came with 8476a2630fb50427e370ca7cbe451d3bf0bfb885.

The change above suggests that `network_config` section belongs to `NetworkAttachmentDefinition` now. This is incorrect. I assume `network_data.yaml` that @hjensas mentioned is the `net_config.yaml` the previous version of the document suggested.

(The change as merged is quite concerning. I think I can't vouch for the document without doing another re-reading from scratch.)

---

I would also rely on @hjensas suggestion that both `openstack overcloud network provision` and `openstack overcloud node provision` are needed. This part of the original instructions were written without validation on a real TripleO cluster due to limited availability of such in adoption test env; where we ended up starting the `standalone` env with the desired static routes preconfigured instead of going through the `network provision` step. So the instruction may be off.